csharp/test/Apache.Arrow.Tests/SchemaBuilderTests.cs

```diff
@@ -16,12 +16,41 @@
 using Apache.Arrow.Types;
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using Xunit;
 
 namespace Apache.Arrow.Tests
 {
     public class SchemaBuilderTests
     {
+        public class SchemaComparer
```

Comment:
I don't think this should be nested in the `SchemaBuilderTests` class, especially since others are going to use it. Can this be a new top-level class and in its own file?